HaveValidSnapshot.swift

```diff
@@ -67,8 +72,15 @@ func _getDefaultReferenceDirectory(sourceFileName: String) -> String {
 
     let pathComponents: NSArray = (sourceFileName as NSString).pathComponents
     for folder in pathComponents {
+        var found = false
```

Comment:
Wouldn't make sense to put this in a `guard` block?

Given that not finding the suffix means «error» or «inability» to continue